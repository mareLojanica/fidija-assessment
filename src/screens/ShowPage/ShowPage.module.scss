.show {
  &__details {
    display: flex;
    gap: 50px;
    width: 100%;
    margin-top: 50px;

    @media (max-width: 768px) {
      flex-direction: column;
      align-items: center;
    }
  }
}

.show-details {
  display: flex;
  gap: 20px;

  &__section {
    flex: 1;
    padding: 20px;

    @media (max-width: 768px) {
      padding: 0;
      max-width: 100%;
    }
  }

  @media (max-width: 768px) {
    flex-direction: column;
  }
}

.show-section {
  background-color: #f9f9f9;
  padding: 20px;
  border-radius: 8px;

  &__title {
    font-size: 20px;
    font-weight: bold;
  }

  &__content {
    display: grid;
    grid-template-columns: 1fr;
    gap: 10px;
    margin-top: 2rem;

    @media (max-width: 768px) {
      grid-template-columns: repeat(2, 1fr);
    }
  }

  &__item {
    display: flex;
    align-items: flex-end;
    padding: 20px 0;
    border-bottom: 1px solid #bfbfbf;
    height: 80px;

    @media (max-width: 768px) {
      border: none;
      justify-content: flex-start;
      align-items: start;
      flex-direction: column;
    }

    strong {
      margin-bottom: 5px;
      text-align: left;
      flex: 1;
    }

    span {
      text-align: left;
      flex: 1;
    }
  }
}

.cast-list {
  margin-top: 2rem;

  &__title {
    font-size: 20px;
    font-weight: bold;
  }

  &__content {
    display: flex;
    flex-direction: column;
    gap: 10px;
  }

  &__item {
    display: flex;
    align-items: center;
    gap: 15px;
    padding: 15px 0;
    border-bottom: 1px solid #bfbfbf;
    height: 80px;

    @media (max-width: 768px) {
      border-bottom: none;
    }
  }

  &__img {
    height: 40px;
    width: 40px;
    border-radius: 50%;
    background-color: #bfbfbf;
  }

  &__details {
    display: flex;
    flex-direction: column;
  }

  &__name {
    font-weight: bold;
  }

  &__character {
    font-size: 12px;
    color: gray;
  }

  &__empty {
    font-size: 14px;
    color: #888;
  }
}

.pagination {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 8px;
  margin-top: 20px;

  &__page {
    padding: 8px 12px;
    background: white;
    border: 1px solid #ddd;
    border-radius: 4px;
    cursor: pointer;
    transition: background 0.3s ease;

    &:hover {
      background: #e0f2ff;
    }
  }

  &__active {
    background: #007bff;
    color: white;
    font-weight: bold;
  }

  &__prev,
  &__next {
    background: white;
    border: 1px solid #ddd;
    border-radius: 4px;
    cursor: pointer;
    transition: background 0.3s ease;
    padding: 8px 12px;

    &:hover {
      background: #e0f2ff;
    }
  }

  &__break {
    padding: 8px 12px;
    color: #777;
  }

  &__link {
    width: 100%;
    height: 100%;
    padding: 3px 8px;
  }

  &__disabled {
    pointer-events: none;
    opacity: 0.5;
    background: #f0f0f0;
    border: 1px solid #ccc;
    color: #999;
    font-weight: normal;

    &:hover {
      background: #f0f0f0;
      color: #999;
    }
  }
}
.placeholder {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 150px;
  font-size: 16px;
  color: #777;
}
