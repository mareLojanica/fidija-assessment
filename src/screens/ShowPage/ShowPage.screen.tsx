import React from "react";
import useShowPage from "../../hooks/useShowPageHook";
import styles from "./ShowPage.module.scss";
import SuspenseLoader from "../../ui-components/SuspenseLoader";
import Header from "../../ui-components/Header";
import Main from "../../ui-components/Main";
import ShowInfoSection from "../../ui-components/ShowInfoSection";
import ShowCastSection from "../../ui-components/ShowCastSection";
import ShowDetails from "../../ui-components/ShowDetails";

const ShowPage: React.FC = () => {
  const { show, isLoading, error } = useShowPage();

  return (
    <>
      <Header
        title="TV Blend"
        showInfo={
          show && (
            <ShowDetails
              image={show.image}
              name={show.name}
              rating={show.rating?.average ?? 0}
              summary={show.summary}
            />
          )
        }
      />
      <Main>
        {isLoading && <SuspenseLoader />}
        {error && (
          <p className={styles["placeholder"]}>Ooops something went wrong!</p>
        )}
        {!isLoading && !error && show ? (
          <div className={styles["show__details"]}>
            <ShowInfoSection
              network={show.network?.name}
              genres={show.genres}
              status={show.status}
              schedule={show.schedule.days}
            />
            <ShowCastSection cast={show?._embedded?.cast ?? []} />
          </div>
        ) : (
          !isLoading &&
          !error && <p className={styles["placeholder"]}>Show not found.</p>
        )}
      </Main>
    </>
  );
};

export default ShowPage;
