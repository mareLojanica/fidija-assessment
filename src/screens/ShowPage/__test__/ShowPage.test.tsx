import React from "react";
import { render, screen } from "@testing-library/react";
import ShowPage from "../ShowPage.screen";
import "@testing-library/jest-dom";
import useShowPage from "../../../hooks/useShowPageHook";

jest.mock("../../../hooks/useShowPageHook", () => ({
  __esModule: true,
  default: jest.fn(),
}));

jest.mock("../../../ui-components/SuspenseLoader", () => () => (
  <div data-testid="suspense-loader">Loading...</div>
));

jest.mock("../../../ui-components/Header", () =>
  jest.fn(({ title }) => <header data-testid="header">{title}</header>)
);

jest.mock("../../../ui-components/Main", () =>
  jest.fn(({ children }) => <main data-testid="main">{children}</main>)
);

jest.mock("../../../ui-components/ShowDetails", () =>
  jest.fn(({ image, name, rating, summary }) => (
    <div data-testid="show-details">
      <img src={image?.original || ""} alt={name} />
      <p>{name}</p>
      <p>{rating}</p>
      <p>{summary}</p>
    </div>
  ))
);

jest.mock("../../../ui-components/ShowInfoSection", () =>
  jest.fn(({ network, genres, status, schedule }) => (
    <div data-testid="show-info-section">
      <p>{network}</p>
      <p>{genres?.join(", ")}</p>
      <p>{status}</p>
      <p>{schedule?.join(", ")}</p>
    </div>
  ))
);

jest.mock("../../../ui-components/ShowCastSection", () =>
  jest.fn(({ cast }) => (
    <div data-testid="show-cast-section">
      {cast.length ? <p>Cast Available</p> : <p>No Cast</p>}
    </div>
  ))
);

describe("ShowPage Component", () => {
  beforeEach(() => {
    (useShowPage as jest.Mock).mockReturnValue({
      show: null,
      isLoading: false,
      error: null,
    });
  });

  test("renders header", () => {
    render(<ShowPage />);
    expect(screen.getByTestId("header")).toBeInTheDocument();
  });

  test("renders loading state", () => {
    (useShowPage as jest.Mock).mockReturnValue({
      show: null,
      isLoading: true,
      error: null,
    });

    render(<ShowPage />);
    expect(screen.getByTestId("suspense-loader")).toBeInTheDocument();
  });

  test("renders error message when there is an error", () => {
    (useShowPage as jest.Mock).mockReturnValue({
      show: null,
      isLoading: false,
      error: "Network Error",
    });

    render(<ShowPage />);
    expect(screen.getByText("Ooops something went wrong!")).toBeInTheDocument();
  });

  test("renders 'Show not found' message when no show is available", () => {
    (useShowPage as jest.Mock).mockReturnValue({
      show: null,
      isLoading: false,
      error: null,
    });

    render(<ShowPage />);
    expect(screen.getByText("Show not found.")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<ShowPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
