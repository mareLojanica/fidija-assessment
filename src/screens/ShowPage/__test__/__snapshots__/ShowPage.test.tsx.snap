// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowPage Component matches snapshot 1`] = `
<DocumentFragment>
  <header
    data-testid="header"
  >
    TV Blend
  </header>
  <main
    data-testid="main"
  >
    <p
      class="placeholder"
      data-testid="show-not-found"
    >
      Show not found.
    </p>
  </main>
</DocumentFragment>
`;
